hcl2template/types.packer_config.go

```diff
@@ -639,6 +642,47 @@ func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceUseBlock, block
 	return res, diags
 }
 
+// GetHCPPackerRegistryBlock return the HCP registry configuration block
+// that can should be used for the current build. Right now, it should
+// use the block at the top level but support the block inside the first
+// build block with a deprecation diagnostic
+func (cfg *PackerConfig) GetHCPPackerRegistry() (*HCPPackerRegistryBlock, hcl.Diagnostics) {
+	var block *HCPPackerRegistryBlock
+	var diags hcl.Diagnostics
+
+	multipleRegistryDiag := func(*HCPPackerRegistryBlock) *hcl.Diagnostic {
+		return &hcl.Diagnostic{
+			Summary:  "Multiple HCP Packer registry block declaration",
+			Subject:  block.HCL2Ref.DefRange.Ptr(),
+			Severity: hcl.DiagError,
+			Detail:   "Multiple HCP Packer Registry block has been found, please configure it at the top level",
```

Comment:
I would suggest being a bit more verbose here for this error
```suggestion
			Detail:   "Multiple HCP Packer Registry blocks have been found, only one can be defined in HCL2 templates. Starting with Packer 1.12.1, it is recommended to move it to the top-level configuration instead of within a build block.",
```